### DISCUSSION TITLE
Keyboard shortcut to toggle lualine (or statusline)

### DISCUSSION THREAD

**Discussion by MAXIIIMVS** [nvim-lualine/lualine.nvim#979] [Q&A] [answered]
Hi.
Is there a way to toggle lualine? I know there's a `hide` method, but I don't know how to check if lualine is present or not.

**Comment by fitrh** [comment, marked as answer]
I think since `hide()` sets `vim.o.statusline` to empty string with `Normal`
highlight, you can try something like this

```lua
vim.api.nvim_set_keymap("n", "<Leader>s", "", {
    callback = function()
        local statusline = vim.o.statusline

        require("lualine").hide({
            place = { "statusline" },
            unhide = statusline == "" or statusline == "%#Normal#",
        })
    end,
})
```

**Reply by MAXIIIMVS** [reply]
Your solution is not completely right, but setting `unhide = vim.o.stautsline == ""` kinda works, but it's not perfect.
Thank you.
